core/trino-main/src/test/java/io/trino/sql/query/TestRowFilter.java

```diff
@@ -405,4 +405,16 @@ public void testUpdate()
         assertThatThrownBy(() -> assertions.query("UPDATE orders SET totalprice = totalprice * 2 WHERE orderkey IN (10, 20, 30)"))
                 .hasMessage("line 1:1: Updating a table with a row filter is not supported");
     }
+
+    @Test
+    public void testInsert()
+    {
+        accessControl.reset();
+        accessControl.rowFilter(
+                new QualifiedObjectName(CATALOG, "tiny", "nation"),
+                USER,
+                new ViewExpression(USER, Optional.empty(), Optional.empty(), "nationkey < 10"));
+        assertThatThrownBy(() -> assertions.query("INSERT INTO nation VALUES (26, 'POLAND', 0, 'No comment')"))
+                .hasMessage("This connector does not support inserts");
```

Comment:
This looks wrong. We should be testing with a connector which does support inserts, or make a mock connector pretend it does that.

Reply:
Let me squash this commit then to avoid imperfect intermediate state.